cmd/gateway-manta.go

```diff
@@ -0,0 +1,368 @@
+package cmd
+
+import (
+	"bytes"
+	"context"
+	"encoding/pem"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path"
+
+	"github.com/hashicorp/errwrap"
+	triton "github.com/joyent/triton-go"
+	"github.com/joyent/triton-go/authentication"
+	"github.com/joyent/triton-go/storage"
+	"github.com/minio/minio-go/pkg/policy"
+)
+
+const defaultMantaURL = "https://us-east.manta.joyent.com"
+const mantaDefaultRootStore = "/stor"
+
+// tritonObjects - Implements Object layer for Triton Manta storage
+type tritonObjects struct {
+	client *storage.StorageClient
+}
+
+func newMantaGateway(host string) (GatewayLayer, error) {
+	var err error
+	var signer authentication.Signer
+	var endpoint = defaultMantaURL
+
+	if host != "" {
+		endpoint, _, err = parseGatewayEndpoint(host)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	creds := serverConfig.GetCredential()
+	keyMaterial := os.Getenv("MINIO_KEY_MATERIAL")
+
+	if keyMaterial == "" {
+		signer, err = authentication.NewSSHAgentSigner(creds.SecretKey, creds.AccessKey)
+		if err != nil {
+			return nil, errwrap.Wrapf("Error Creating SSH Agent Signer: {{err}}", err)
```

Comment:
You don't need to use errwrap.Wrapf instead, do `traceError(err)` recommended here..